src/devices/cpu/dsp56156/dsp56mem.cpp

```diff
@@ -53,6 +53,7 @@ void HCR_set(dsp56156_core* cpustate, uint16_t value)
 	HCIE_bit_set(cpustate, (value & 0x0004) >> 2);
 	HTIE_bit_set(cpustate, (value & 0x0002) >> 1);
 	HRIE_bit_set(cpustate, (value & 0x0001) >> 0);
+	cpustate->device->machine().scheduler().synchronize();
```

Comment:
This will try to end the timeslice, but the changes to the HCR bits will be visible immediately.  Do you want to defer making them visible until the other CPU(s) have caught up?  That would require using a callback here.

Reply:
This was an oversight. Addressing.

Reply:
Removed it entirely, as it was from a failed attempt at debugging why there appeared to be issues with buffer-flipping.